l2geth/core/vm/evm.go

```diff
@@ -91,6 +91,9 @@ type Context struct {
 	BlockNumber *big.Int       // Provides information for NUMBER
 	Time        *big.Int       // Provides information for TIME
 	Difficulty  *big.Int       // Provides information for DIFFICULTY
+
+	// OVM information
+	L1MessageSender common.Address // Provides information for L1MESSAGESENDER
 }
```

Comment:
Longer term we are thinking of also adding `L1Timestamp` and `L1BlockNumber` or will those stay as modifications of `NUMBER` and `TIMESTAMP`? :thinking:

Reply:
Personally I would like to add L1Timestamp and L1BlockNumber here but we need to have that convo with the rest of the team